ledger/fullblock_perf_test.go

```diff
@@ -0,0 +1,331 @@
+	_, err = rand.Read(creator[:])
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+	genesisInitState.Accounts[creator] = basics.MakeAccountData(basics.Offline, basics.MicroAlgos{Raw: 12345678900000})
+	for i := 0; i < numAccts; i++ {
+		acct := basics.Address{}
+		_, err = rand.Read(acct[:])
+		if err != nil {
+			return nil, nil, basics.Address{}, nil, nil, err
+		}
+		genesisInitState.Accounts[acct] = basics.MakeAccountData(basics.Offline, basics.MicroAlgos{Raw: 1234567890})
+		accts = append(accts, acct)
+	}
+
+	// open ledger
+	const inMem = false
+	cfg := config.GetDefaultLocal()
+	cfg.Archival = true
+	l0, err = OpenLedger(logging.Base(), dbPrefix, inMem, genesisInitState, cfg)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+
+	// open second ledger
+	dbName = fmt.Sprintf("%s.%d.2", name, crypto.RandUint64())
+	dbPrefix = filepath.Join(dbTempDir, dbName)
+	l1, err = OpenLedger(logging.Base(), dbPrefix, inMem, genesisInitState, cfg)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+
+	blk := genesisInitState.Block
+	blk.BlockHeader.Round++
+	blk.BlockHeader.TimeStamp += int64(crypto.RandUint64() % 100 * 1000)
+	blk.BlockHeader.GenesisID = "x"
+	cert := agreement.Certificate{}
+
+	err = l0.AddBlock(blk, cert)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+	err = l1.AddBlock(blk, cert)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+
+	newBlk := bookkeeping.MakeBlock(blk.BlockHeader)
+	eval, err := l0.StartEvaluator(newBlk.BlockHeader, 5000, 0)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+
+	for i, acct := range accts {
+		tx := createAssetTransaction(uint64(i), 1, creator, crypto.Digest{1})
+		stxn := transactions.SignedTxn{Txn: tx, Sig: crypto.Signature{1}}
+		err = eval.Transaction(stxn, transactions.ApplyData{})
+		if err != nil {
+			return nil, nil, basics.Address{}, nil, nil, err
+		}
+
+		// opt in to the asset
+		aIdx := eval.TestingTxnCounter()
+		tx = sendAssetTransaction(uint64(i), 1, acct, 0, crypto.Digest{1}, basics.AssetIndex(aIdx), acct)
+		aIdxs = append(aIdxs, aIdx)
+		stxn = transactions.SignedTxn{Txn: tx, Sig: crypto.Signature{1}}
+		err = eval.Transaction(stxn, transactions.ApplyData{})
+		if err != nil {
+			return nil, nil, basics.Address{}, nil, nil, err
+		}
+
+	}
+
+	vblk, err := eval.GenerateBlock()
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+	err = l0.AddBlock(vblk.Block(), cert)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+	err = l1.AddBlock(vblk.Block(), cert)
+	if err != nil {
+		return nil, nil, basics.Address{}, nil, nil, err
+	}
+
+	return l0, l1, creator, accts, aIdxs, nil
+}
+
+func BenchmarkBlockValidationMix(b *testing.B) {
+	numAssets := 10000
+	benchmarkBlockValidationMix(b, numAssets)
+}
+
+func BenchmarkBlockValidationPayments(b *testing.B) {
+	benchmarkBlockValidationMix(b, 0)
```

Comment:
when we pass 0 to setupEnv, it does not create any accounts, except a single creator address, that is not great.
I suggest to have say 50k addresses in a pool for both asset and payment txns, and have a random function for payment receiver: 25% it is a new random address, 75% some existing address.